File: 2026_05/27_Pizza_Party/test_pizza_party.py
from main_pizza_party import get_pizzas_to_order as gpo

# Test ----------------------------------------------------

print("Test 1: --------------------------------------------")
print(f"Ergebnis: {gpo([8, 8, 8])}")
print("Erwartet: 2")

print("Test 2: --------------------------------------------")
print(f"Ergebnis: {gpo([10, 9, 8, 2, 2, 6, 10])}")
print("Erwartet: 3")

print("Test 3: --------------------------------------------")
print(f"Ergebnis: {gpo([1, 2, 3, 4, 5])}")
print("Erwartet: 2")

print("Test 4: --------------------------------------------")
print(f"Ergebnis: {gpo([8, 8, 8, 8, 8, 8, 8, 8])}")
print("Erwartet: 3")

print("Test 5: --------------------------------------------")
print(f"Ergebnis: {gpo([9, 9, 6])}")
print("Erwartet: 1")

print("Test 6: --------------------------------------------")
print(f"Ergebnis: {gpo([10, 12, 16, 9, 8, 11, 15, 8, 0])}")
print("Erwartet: 5")

File: 2026_05/27_Pizza_Party/test_pizza_party.js
import { getPizzasToOrder } from "./main_pizza_party.js";

/*
1. getPizzasToOrder([8, 8, 8]) should return 2.
2. getPizzasToOrder([10, 9, 8, 2, 2, 6, 10]) should return 3.
3. getPizzasToOrder([1, 2, 3, 4, 5]) should return 2.
4. getPizzasToOrder([8, 8, 8, 8, 8, 8, 8, 8]) should return 3.
5. getPizzasToOrder([9, 9, 6]) should return 1.
6. getPizzasToOrder([10, 12, 16, 9, 8, 11, 15, 8, 0]) should return 5.
*/

// Tests ------------------------------------------------------------

console.log("Test 1: --------------------------------------")
console.log(`Result: ${getPizzasToOrder([8, 8, 8])}`);
console.log("Expected: 2");

console.log("Test 2: --------------------------------------")
console.log(`Result: ${getPizzasToOrder([10, 9, 8, 2, 2, 6, 10])}`);
console.log("Expected: 3");

console.log("Test 3: --------------------------------------")
console.log(`Result: ${getPizzasToOrder([1, 2, 3, 4, 5])}`);
console.log("Expected: 2");

console.log("Test 4: --------------------------------------")
console.log(`Result: ${getPizzasToOrder([8, 8, 8, 8, 8, 8, 8, 8])}`);
console.log("Expected: 3");

console.log("Test 5: --------------------------------------")
console.log(`Result: ${getPizzasToOrder([9, 9, 6])}`);
console.log("Expected: 1");

console.log("Test 6: --------------------------------------")
console.log(`Result: ${getPizzasToOrder([10, 12, 16, 9, 8, 11, 15, 8, 0])}`);
console.log("Expected: 5");
